exams/pearson/readers.py

```diff
@@ -0,0 +1,100 @@
+"""
+Readers for data Pearson exports
+"""
+import csv
+from datetime import datetime
+
+import pytz
+
+from exams.pearson.constants import (
+    PEARSON_CSV_DIALECT,
+    PEARSON_DATETIME_FORMAT,
+)
+from exams.pearson.exceptions import InvalidTsvRowException
+
+
+class BaseTSVReader(object):
+    """
+    Base class for TSV file readers
+
+    It handles the high-level mapping and reading of the data.
+    Subclasses specify how the fields map.
+    """
+    def __init__(self, field_mappers, read_as_cls):
+        """
+        Initializes a new TSV writer
+
+        The first value of each fields tuple is the destination field name.
+        The second value is a str property path (e.g. "one.two.three") or
+        a callable that when passed a row returns a computed field value
+
+        Usage:
+
+            # maps column 'ABC' to field 'abc' and casts it to an int
+            # maps column 'DEF' to field 'def' and leaves it as a str
+            # Initializes each row into the namedtuple
+            Record = namedtuple('Record', ['abc', 'def'])
+
+            BaseTSVReader([
+                ('ABC', 'abc', int),
+                ('DEF', 'def'),
+            ], Record)
+
+        Arguments:
+            field_mappers (list(tuple)): list of tuple field mappers
+            read_as_cld (cls): class to instantiate row data
```

Comment:
I believe generally we follow this which says either way is acceptable: http://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_google.html

In practice, so far we're using the style used in this PR [here](https://github.com/mitodl/micromasters/blob/bf1ec5f0e9c38855190288b895ccf215e8c5b062/dashboard/api_edx_cache.py#L30) and [here](https://github.com/mitodl/micromasters/blob/4b000f07106e814bfe546cb4b92e55dbaad7403f/dashboard/utils.py#L43) so I think if this is something we want to change, that's fair for discussion, but I don't think in the context of this PR.